apisix/plugins/proxy-rewrite.lua

```diff
@@ -153,10 +154,20 @@ function _M.rewrite(conf, ctx)
         end
     end
 
-    upstream_uri = core.utils.uri_safe_encode(upstream_uri)
+    local index = find_str(upstream_uri, "?", 1, true)
+    if index then
+        upstream_uri = core.utils.uri_safe_encode(sub_str(upstream_uri, 1, index-1)) ..
+                       sub_str(upstream_uri, index)
+    else
+        upstream_uri = core.utils.uri_safe_encode(upstream_uri)
+    end
 
     if ctx.var.is_args == "?" then
-        ctx.var.upstream_uri = upstream_uri .. "?" .. (ctx.var.args or "")
+        if index then
+            ctx.var.upstream_uri = upstream_uri .. "&" .. (ctx.var.args or "")
```

Comment:
I'm wondering if we need to keep the request args?

Reply:
> I'm wondering do we need to keep the request args?

I think we need to keep it.
If someone needn't it, he can remove args by regex_uri.